packages/run-protocol/src/attestation/attestation.js

```diff
@@ -0,0 +1,335 @@
+// @ts-check
+
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { E } from '@agoric/eventual-send';
+import { Far } from '@endo/marshal';
+
+import { getCopyBagEntries, makeCopyBag, makeStore } from '@agoric/store';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+import { assertPrerequisites } from './prerequisites.js';
+
+const { details: X } = assert;
+
+/**
+ * Add a lien to an amount for an address.
+ *
+ * @param {Store<Address,Amount>} store
+ * @param {Address} address
+ * @param {Amount} amountToLien
+ */
+const addToLiened = (store, address, amountToLien) => {
+  let updated;
+  if (store.has(address)) {
+    updated = AmountMath.add(store.get(address), amountToLien);
+    store.set(address, updated);
+  } else {
+    updated = amountToLien;
+    store.init(address, amountToLien);
+  }
+  return updated;
+};
+harden(addToLiened);
+
+/**
+ * Validate that the address is a string and that the amount is a
+ * valid Amount of the right brand. Coerce the amount and return it.
+ *
+ * @param {Brand} externalBrand
+ * @param {Address} address
+ * @param {Amount} amount
+ * @returns {Amount}
+ */
+const validateInputs = (externalBrand, address, amount) => {
+  assert.typeof(address, 'string');
+  // Will throw if amount is invalid
+  return AmountMath.coerce(externalBrand, amount);
+};
+harden(validateInputs);
+
+/**
+ * @param {ContractFacet} zcf
+ * @param {ZCFMint} zcfMint
+ * @param {Amount} amountToMint
+ * @returns {Promise<Payment>}
+ */
+const mintZCFMintPayment = (zcf, zcfMint, amountToMint) => {
```

Comment:
I don't know. It could be left over from when there were to kinds of attestations. I got rid of the timed attestation code and tested that the remaining stuff passes its tests but I haven't gotten as far as I'd like getting down to only the essential code.

Reply:
I figured it out and added a comment:

```
 * To support `makeAttestation`, we directly mint attestation payments.
 * We still use a ZCFMint because returning attestations is done
 * through Zoe offers.
```